pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -254,10 +254,12 @@ func (p *PoolManager) allocateRequestedVirtualMachineInterfaceMac(virtualMachine
 	}
 
 	if _, exist := p.macPoolMap[requestedMac]; exist {
-		err := fmt.Errorf("failed to allocate requested mac address")
-		logger.Error(err, "mac address already allocated")
-
-		return err
+		// check if mac already belongs to the vm
+		if macFromVmMap, _ := p.vmToMacPoolMap[vmNamespaced(virtualMachine)][iface.Name]; macFromVmMap != requestedMac {
```

Comment:
perhaps I should check it sooner, using `vmToMacPoolMap` to look for allocated macs instead of the previos vm spec.
@qinqon what do you think?